src/OrbitGl/App.cpp

```diff
@@ -1083,8 +1084,34 @@ void OrbitApp::StartCapture() {
 
   CHECK(capture_client_ != nullptr);
 
-  auto capture_event_processor =
-      CaptureEventProcessor::CreateForCaptureListener(this, std::move(frame_track_function_ids));
+  std::unique_ptr<CaptureEventProcessor> capture_event_processor;
+  auto capture_for_listener_processor =
+      CaptureEventProcessor::CreateForCaptureListener(this, frame_track_function_ids);
+
+  if (absl::GetFlag(FLAGS_enable_capture_autosave)) {
+    std::filesystem::path file_path =
+        Path::CreateOrGetCaptureDir() /
+        orbit_client_model::capture_serializer::GenerateCaptureFileName(process->name(),
+                                                                        absl::Now(), "_autosave");
+    auto save_to_file_processor_or_error = CaptureEventProcessor::CreateSaveToFileProcessor(
+        file_path, frame_track_function_ids, [this](const ErrorMessage& error) {
+          SendErrorToUi("Error saving capture", error.message());
+        });
+    if (save_to_file_processor_or_error.has_value()) {
+      std::vector<std::unique_ptr<CaptureEventProcessor>> event_processors;
+      event_processors.push_back(std::move(capture_for_listener_processor));
+      event_processors.push_back(std::move(save_to_file_processor_or_error.value()));
+      capture_event_processor =
+          CaptureEventProcessor::CreateCompositeProcessor(std::move(event_processors));
+    } else {
+      SendErrorToUi("Error saving capture",
+                    absl::StrFormat("Unable to save capture to \"%s\": %s", file_path.string(),
```

Comment:
This error might be a confusing for the user. This error potentially shows up on capture start, correct? Then maybe add some more context like: "Unable to set up automatic capture saving"